haskell-src/exec/Chainweb/Listen.hs

```diff
@@ -67,8 +70,18 @@ getOutputsAndInsert env ph@(PowHeader h _) = do
              (hashB64U $ _blockHeader_hash h)
       logg Info $ fromString $ show e
     Right pl -> do
+      let chain = _blockHeader_chainId h
+          height = _blockHeader_height h
+          hash = _blockHeader_hash h
+          tos = _blockPayloadWithOutputs_transactionsWithOutputs pl
+          ts = S.fromList $ map (\(t,tout) -> mkTxSummary chain height hash t tout) tos
+          msg = printf "Got new header on chain %d height %d" (unChainId chain) height
+          addendum = if S.null ts then "" else printf " with %d transactions" (S.length ts)
+      when logTxSummaries $ do
+        logg Info $ fromString $ msg <> addendum
+        mapM_ (logg Info . fromString . show) ts
       insertNewHeader (_nodeInfo_chainwebVer $ _env_nodeInfo env) (_env_dbConnPool env) ph pl
-      logg Info (fromString $ printf "%d" (unChainId $ _blockHeader_chainId h)) >> hFlush stdout
+      logg Info (fromString $ printf "%d" (unChainId chain)) >> hFlush stdout
```

Comment:
Seems like this log line will also get printed in `server` mode and it will be redundant since we already have the `Got new header on chain %d height %d` logs in that case.